### ISSUE TITLE
Update `settings.gradle` for example apps to remove `.flutter-plugins`

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#157660]
### Overview
Blocks https://github.com/flutter/flutter/issues/48918.

See [`repo:flutter/packages /.flutter-plugins[^-]/`](https://github.com/search?q=repo%3Aflutter%2Fpackages%20%2F.flutter-plugins%5B%5E-%5D%2F&type=code); there are a number of example apps (or, all of the Android ones) that still rely on the 2019-era `.flutter-plugins` file, and should be upgraded to use `.flutter-plugins-dependencies`.

See https://docs.flutter.dev/release/breaking-changes/flutter-plugins-configuration for instructions.

I am not sure what the best way is to recreate the build scripts across these repos, and would love some advice from either @reidbaker or @stuartmorgan, but I'm happy to do the work once back from OOO, or if @camsim99 feels like pinching in :)

/cc @gmackall as well.

### Progress Tracker

| Package | Owner | PR
| --------- | ------ | --- |
| animations  | @camsim99  | https://github.com/flutter/packages/pull/8002 |
| camera | @jesswrd |  |
| camera_android | @jesswrd |  |
| espresso | ~@camsim99~  @jesswrd | ~https://github.com/flutter/packages/pull/8003~ https://github.com/flutter/packages/pull/8037 |
| google_maps_flutter | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| google_maps_flutter_android | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| google_sign_in |@jesswrd | https://github.com/flutter/packages/pull/8019 |
| google_sign_in_android | @jesswrd | https://github.com/flutter/packages/pull/8019 |
| image_picker | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| image_picker_android | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| in_app_purchase | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| in_app_purchase_android | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| local_auth | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| local_auth_android | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| palette_generator | @jesswrd | https://github.com/flutter/packages/pull/8019 |
| path_provider_android | @jesswrd | https://github.com/flutter/packages/pull/7822 |
| quick_actions | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| quick_actions_android | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| url_launcher | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| url_launcher_android | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| video_player | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| video_player_android | @jesswrd | https://github.com/flutter/packages/pull/8037 |
| webview_flutter | @jewsswrd | https://github.com/flutter/packages/pull/7968 |
| webview_flutter_android | @jesswrd | https://github.com/flutter/packages/pull/7968 |

Notes:
- We will also need to modify https://github.com/flutter/packages/blob/1d00782b90d4c57f03c7b6b1b4e4f91d39a0426b/script/tool/test/gradle_check_command_test.dart#L190
- If this list misses any Android example apps, we need to file another issue to finish any leftover Kotlin conversions; see https://github.com/flutter/flutter/issues/157660#issuecomment-2441853854

**Comment by reidbaker** [comment]
Since we have custom artifactory handling I dont think there is an automated way. I think you would just delete the section you have and then probably update tooling that verifies we have this set up correctly.

**Comment by stuartmorgan** [comment]
> I dont think there is an automated way.

Agreed, I think any mass update here will have to be manual, or a bespoke script.

Something we could consider if we're going to have to do a bunch of manual work here anyway: we could do a tech debt reset on the Android example apps and instead of updating this specific thing, instead make a script that deletes the `android` directory from all the examples and then re-`flutter create`s it (ideally using the Kotlin template, as that's presumably what the vast majority of our users are using, and we are currently not testing), and then manually re-apply intentional changes like adding permissions where necessary, and adding the warning settings in the example build apps.

It might not be much more work, and it would catch us up across the board. We don't need to keep old apps on a per-package basis, because we have explicit legacy app testing as part of build-all.

@reidbaker Thoughts?

**Comment by reidbaker** [comment]
I like the idea but worry about missing some of those intentional bits accidentally. I defer to @camsim99 but given how simple of a migration this specific one should be I would do what you are suggesting independent of this task.

**Comment by gmackall** [comment]
FYI that we do have a bespoke script for the framework repo 
https://github.com/flutter/flutter/blob/b98d9a37ac9807fe387d3b616c31e37197948013/dev/tools/bin/generate_gradle_lockfiles.dart#L260
(not sure if you need to do updates there though or if it is all in the packages repo, also it does a little more than just overwrite gradle files)

**Comment by reidbaker** [comment]
That is weird because the string there does not reference the plugin file that this bug is about.

**Comment by gmackall** [comment]
> That is weird because the string there does not reference the plugin file that this bug is about.

Hmm, you know I should have checked this, because it means all the `settings.gradle`s managed by that script are obviously updated!

**Comment by matanlurey** [comment]
Thanks for the discussion while I was away.

I don't have a strong opinion, I could see us either:
1. Regenerating the `example/` folder (Stuart's original suggestion)
1. Making surgical fixes to this file in particular, likely manually
1. Regenerating this file in particular across the repo (similar to the framework)

Of all the suggestions, I prefer (1) - I think all of them have the chance to cause a bit of toil, and only (1) leaves us at probably the healthiest end-state. If there isn't a strong push back, I'll consider this decided and start working on it.

**Comment by stuartmorgan** [comment]
FYI if you go with 1, a couple of tips:
- Make sure you are on `stable` (not `master`) when you do it, as we don't guarantee backward compatibility of projects, and all the examples need to build and run on `stable`.
- You'll often need to see what the org was and explicitly pass it as an argument; they were unfortunately not created uniformly, as I discovered when adding desktop support to examples. (I think it's always the same as the start of the `namespace` in example/android/app/build.gradle.)

**Comment by matanlurey** [comment]
Ack, thanks!

**Comment by matanlurey** [comment]
@camsim99 I see you were assigned by Reid, I hope this is a short/fast change - LMK if you could use help!

**Comment by camsim99** [comment]
Oop except one thing -- we can remove backwards compatibility from the plugins tool once that PR lands: https://github.com/flutter/packages/blob/7b6583c8dc0efb8edac3bc1f7e4f5db9fba19c3e/script/tool/test/gradle_check_command_test.dart#L190
